src/components/DatePickerDocked/HeaderItem.tsx

```diff
@@ -0,0 +1,117 @@
+
+    const {
+        buttonContainerStyle,
+        buttonWrapperStyle,
+        spacerStyle,
+        buttonStyle,
+        innerStyle,
+        labelStyle,
+        iconContainerStyle,
+        emtpyViewStyle,
+    } = useMemo(() => {
+        const {
+            buttonContainer,
+            buttonWrapper,
+            spacer,
+            buttonStyle: _buttonStyle,
+            innerStyle: _innerStyle,
+            labelStyle: _labelStyle,
+            emtpyView,
+        } = headerItemStyles;
+
+        return {
+            buttonContainerStyle: buttonContainer,
+            buttonWrapperStyle: buttonWrapper,
+            spacerStyle: spacer,
+            buttonStyle: _buttonStyle,
+            innerStyle: _innerStyle,
+            labelStyle: _labelStyle,
+            iconContainerStyle: { opacity: 1 },
+            emtpyViewStyle: emtpyView,
+        };
+    }, [headerItemStyles]);
+
+    const handlePressDropDown = useCallback(() => {
+        onPressDropdown(type);
+    }, [onPressDropdown, type]);
+
+    return (
+        <View style={buttonContainerStyle} pointerEvents={'box-none'}>
+            <View style={spacerStyle} pointerEvents={'box-none'} />
+            {!selecting ? (
+                <View style={buttonWrapperStyle}>
+                    <IconButton
+                        type="material-community"
+                        name="chevron-left"
+                        size="sm"
+                        // Todo: Translate
+                        accessibilityLabel={'Previous'}
+                        onPress={onPrev}
+                        disabled={value === 1800}
+                    />
+                </View>
+            ) : (
+                <View style={emtpyViewStyle} />
+            )}
+            <TouchableRipple
+                disabled={disabled}
+                onPress={handlePressDropDown}
+                accessibilityRole="button"
+                accessibilityLabel={`${value}`}
+                style={buttonStyle}>
+                <View style={innerStyle}>
+                    <Text style={labelStyle} selectable={false}>
+                        {value}
+                    </Text>
+                    <View style={iconContainerStyle}>
+                        <IconButton
+                            onPress={handlePressDropDown}
+                            name={selecting ? 'menu-up' : 'menu-down'}
+                            size="xs"
+                            disabled={disabled}
+                        />
+                    </View>
+                </View>
+            </TouchableRipple>
+            {!selecting ? (
+                <View style={buttonWrapperStyle}>
+                    <IconButton
+                        name="chevron-right"
+                        size="sm"
+                        // Todo: Translate
+                        accessibilityLabel={'Next'}
+                        onPress={onNext}
+                        disabled={value === 2200}
```

Comment:
hardcoded values are not good. user could define startYear to be 2000 and endYear to be 2010 and yet the user can go upto 2200.. which is incorrect.

Reply:
done @shashank1010

Reply:
using endYear prop